src/ASM/ASMbase.C

```diff
@@ -989,6 +989,29 @@ void ASMbase::extractNodeVec (const Vector& globRes, Vector& nodeVec,
   }
 }
 
+void ASMbase::injectNodeVec (const std::vector<int>& madof,
+                             const Vector& nodeVec, Vector& globVec,
+                             int basis) const
+{
+  size_t ldof = 0;
+  char bType = basis == 1 ? 'D' : 'P'+basis-2;
+  for (size_t i = 0; i < MLGN.size(); i++)
+  {
+    if (basis == 0 || getNodeType(i+1) == bType) {
+      int inod = MLGN[i];
+      int idof = madof[inod-1] - 1;
+      int jdof = madof[inod] - 1;
+#ifdef INDEX_CHECK
+      if (inod < 1 || jdof > (int)globVec.size())
+        std::cerr <<" *** ASMbase::injectNodeVec: Global DOF "<< jdof
+                  <<" is out of range [1,"<< globVec.size() <<"]"<< std::endl;
+#endif
+      std::copy(nodeVec.begin()+ldof, nodeVec.begin()+ldof+(jdof-idof), globVec.begin()+idof);
```

Comment:
Den sjekken med melding blir litt feil hvis inod er out-of-bound fordi jdof da kan bli noe tilfeldig. Og copy-statementet bør vel ikke utføres hvis idof eller jdof er ugyldig (tilsvarende logikk som i den andre metoden).

Dessuten (kosmetisk): brace-par på linje 999,1012 er overflødig da kun ett statement i løkka (if-statementet). this-> på getNodeType kallet.